Traceflow unit test "TestTraceflow/normalTraceflow" is flaky

**Describe the bug**
<!--
A clear and concise description of what the bug is.
If you believe this bug is a security issue, please don't use this template and follow our [security guidelines](/SECURITY.md)
-->
This unit test is flaky, it might pass if we re-run it after the failure. I got the following error output it failed.

```
I1118 10:57:50.911916   15845 controller.go:138] Starting TraceflowController
I1118 10:57:50.912252   15845 shared_informer.go:240] Waiting for caches to sync for TraceflowController
I1118 10:57:50.912334   15845 shared_informer.go:247] Caches are synced for TraceflowController 
I1118 10:57:50.912469   15845 controller.go:170] Processing Traceflow tf1 ADD event
E1118 10:57:50.913601   15845 controller.go:273] "Invalid Traceflow request" err="requested source Pod ns1/pod1 not found" request=&{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:tf1 GenerateName: Namespace: SelfLink: UID:uid1 ResourceVersion: Generation:0 CreationTimestamp:2021-11-18 10:57:50.911385677 +0000 UTC m=+0.168847152 DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[]} Spec:{Source:{Namespace:ns1 Pod:pod1 IP:} Destination:{Namespace:ns2 Pod:pod2 Service: IP:} Packet:{SrcIP: DstIP: Length:0 IPHeader:{SrcIP: Protocol:0 TTL:0 Flags:0} IPv6Header:<nil> TransportHeader:{ICMP:<nil> UDP:<nil> TCP:<nil>}} LiveTraffic:false DroppedOnly:false Timeout:2} Status:{Phase: Reason: StartTime:<nil> DataplaneTag:0 Results:[] CapturedPacket:<nil>}}
I1118 10:57:50.913870   15845 controller.go:176] Processing Traceflow tf1 UPDATE event
--- FAIL: TestTraceflow (1.01s)
    --- FAIL: TestTraceflow/normalTraceflow (1.00s)
        controller_test.go:105: 
            	Error Trace:	controller_test.go:105
            	Error:      	Expected value not to be nil.
            	Test:       	TestTraceflow/normalTraceflow
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1e33f30]

goroutine 26 [running]:
testing.tRunner.func1.2({0x1fc6c80, 0x31d34a0})
	/opt/hostedtoolcache/go/1.17.3/x64/src/testing/testing.go:1209 +0x36c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.17.3/x64/src/testing/testing.go:1212 +0x3b6
panic({0x1fc6c80, 0x31d34a0})
	/opt/hostedtoolcache/go/1.17.3/x64/src/runtime/panic.go:1047 +0x266
antrea.io/antrea/pkg/controller/traceflow.TestTraceflow.func2(0x0)
	/home/runner/work/antrea/antrea/pkg/controller/traceflow/controller_test.go:107 +0x2b0
testing.tRunner(0xc00043f860, 0xc0003db380)
	/opt/hostedtoolcache/go/1.17.3/x64/src/testing/testing.go:1259 +0x230
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.17.3/x64/src/testing/testing.go:1306 +0x727
```

**To Reproduce**
<!-- Steps to reproduce the behavior. -->

**Expected**
<!-- A clear and concise description of what you expected to happen. -->

**Actual behavior**
<!-- A clear and concise description of what's the actual behavior. If applicable, add screenshots, log messages, etc. to help explain the problem. -->

**Versions:**
<!--
Please provide the following information:
 - Antrea version (Docker image tag).
 - Kubernetes version (use `kubectl version`). If your Kubernetes components have [different versions](https://kubernetes.io/docs/setup/release/version-skew-policy/), please provide the version for all of them.
 - Container runtime: which runtime are you using (e.g. containerd, cri-o, docker) and which version are you using?
 - Linux kernel version on the Kubernetes Nodes (`uname -r`).
 - If you chose to compile the Open vSwitch kernel module manually instead of using the kernel module built into the Linux kernel, which version of the OVS kernel module are you using? Include the output of `modinfo openvswitch` for the Kubernetes Nodes.
-->
Anrea 1.4

**Additional context**
<!-- Add any other context about the problem here, such as Antrea logs, kubelet logs, etc. -->

<!-- (Please consider pasting long output into a [GitHub gist](https://gist.github.com) or any other pastebin.) -->

--- a/pkg/controller/traceflow/controller_test.go
+++ b/pkg/controller/traceflow/controller_test.go
@@ -21,6 +21,7 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
@@ -100,9 +101,11 @@ func TestTraceflow(t *testing.T) {
 		}
 
 		tfc.kubeClient.CoreV1().Pods("ns1").Create(context.TODO(), &pod1, metav1.CreateOptions{})
+		createdPod, _ := tfc.waitForPodInNamespace("ns1", "pod1", time.Second)
+		require.NotNil(t, createdPod)
 		tfc.client.CrdV1alpha1().Traceflows().Create(context.TODO(), &tf1, metav1.CreateOptions{})
 		res, _ := tfc.waitForTraceflow("tf1", crdv1alpha1.Running, time.Second)
-		assert.NotNil(t, res)
+		require.NotNil(t, res)
 		// DataplaneTag should be allocated by Controller.
 		assert.True(t, res.Status.DataplaneTag > 0)
 		assert.Equal(t, numRunningTraceflows(), 1)
@@ -158,15 +161,33 @@ func TestTraceflow(t *testing.T) {
 		}
 
 		tfc.kubeClient.CoreV1().Pods("ns1").Create(context.TODO(), &pod2, metav1.CreateOptions{})
+		createdPod, _ := tfc.waitForPodInNamespace("ns1", "pod2", time.Second)
+		require.NotNil(t, createdPod)
 		tfc.client.CrdV1alpha1().Traceflows().Create(context.TODO(), &tf2, metav1.CreateOptions{})
 		res, _ := tfc.waitForTraceflow("tf2", crdv1alpha1.Failed, time.Second)
-		assert.NotNil(t, res)
+		require.NotNil(t, res)
 		// DataplaneTag should not be allocated by Controller.
 		assert.True(t, res.Status.DataplaneTag == 0)
 		assert.Equal(t, numRunningTraceflows(), 0)
 	})
 
 	close(stopCh)
+}
+
+func (tfc *traceflowController) waitForPodInNamespace(ns string, name string, timeout time.Duration) (*corev1.Pod, error) {
+	var pod *corev1.Pod
+	var err error
+	if err = wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		// Make sure dummy Pod is synced by informer
+		pod, err = tfc.podLister.Pods(ns).Get(name)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+	return pod, nil
 }
 
 func (tfc *traceflowController) waitForTraceflow(name string, phase crdv1alpha1.TraceflowPhase, timeout time.Duration) (*crdv1alpha1.Traceflow, error) {

